resources/assets/components/Post/index.js

```diff
@@ -82,22 +66,24 @@ class Post extends React.Component {
       <div className="post container__row">
         {/* Post Images */}
         <div className="container__block -third images">
-          {this.state.loading ?
-            <div className="is-loading">
-              <div className="spinner"></div>
-            </div>
-          :
-            <img className="post__image" src={getEditedImageUrl(post)}/>
-          }
+          <div className="post__image">
+            <img src={getImageUrlFromPost(post, 'edited')}/>
+          </div>
           <div className="admin-tools">
             <div className="admin-tools__links">
-              <a href={getImageUrlFromProp(post)} target="_blank">Original Photo</a>
+              <a href={getImageUrlFromPost(post, 'original')} target="_blank">Original Photo</a>
               <br />
-              <a href={getEditedImageUrl(post)} target="_blank">Edited Photo</a>
-            </div>
-            <div className="admin-tools__rotate">
-              <a className="button -tertiary rotate" onClick={(event) => this.rotate(event) }></a>
+              <a href={getImageUrlFromPost(post, 'edited')} target="_blank">Edited Photo</a>
             </div>
+            {this.props.rotate ?
+              <div className="admin-tools__rotate">
+                {this.state.loading ?
+                  <div className="spinner"></div>
+                :
+                  <a className="button -tertiary rotate" onClick={(event) => this.handleClick(event)}></a>
```

Comment:
If you want, this could be simplified to `onClick={this.handleClick}` 🐭

Reply:
:) !!!!